meerkat/model/model.py

```diff
@@ -0,0 +1,127 @@
+from typing import Dict, List
+
+import torch
+
+from meerkat import DataPanel
+from meerkat.columns.prediction_column import ClassificationOutputColumn
+from meerkat.model.metrics import compute_metric
+
+
```

Comment:
Something to consider for a future PR: I think that a lot of the functionality implemented in the `Model` class and it's subclasses (`HuggingFaceModel`, `TensorModel`) could be implemented in a functional API. We can discuss this for a future PR, but this looks good for this PR.